### ISSUE TITLE
SLHCUpgradeSimulations/Geometry unit test failure 

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#29474]
There is a failing unit test in `SLHCUpgradeSimulations/Geometry` since `CMSSW_11_1_X_2020-04-14-2300`, after the integration of 
```
===== Test "testPhase2PixelNtuple" ====
argument 0: testPhase2PixelNtuple
argument 1: /bin/bash
argument 2: SLHCUpgradeSimulations/Geometry/test
argument 3: phase2_digi_reco_pixelntuple.sh
shell is: /bin/bash
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_11_1_X_2020-04-14-2300
topdir is: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_11_1_X_2020-04-14-2300
testdir is: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_11_1_X_2020-04-14-2300/src/SLHCUpgradeSimulations/Geometry/test
tmpdir is: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_11_1_X_2020-04-14-2300/tmp/cc8_amd64_gcc8
testbin is: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_11_1_X_2020-04-14-2300/test/cc8_amd64_gcc8
Running script: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_11_1_X_2020-04-14-2300/src/SLHCUpgradeSimulations/Geometry/test/phase2_digi_reco_pixelntuple.sh
15-Apr-2020 04:00:49 CEST  Initiating request to open file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_10_6_0_patch2/RelValSingleMuPt10/GEN-SIM/106X_upgrade2023_realistic_v3_2023D41noPU-v1/10000/7377ED92-245C-CC4D-9F05-25ABB5522A08.root
15-Apr-2020 04:00:56 CEST  Successfully opened file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_10_6_0_patch2/RelValSingleMuPt10/GEN-SIM/106X_upgrade2023_realistic_v3_2023D41noPU-v1/10000/7377ED92-245C-CC4D-9F05-25ABB5522A08.root
2020-04-15 04:02:09.937288: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
----- Begin Fatal Exception 15-Apr-2020 04:03:37 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Processing  stream begin Run run: 1 stream: 0
   [1] Calling method for module trackerDTC::Producer/'TrackerDTCProducer'
   [2] trackerDTC::Settings::checkConfiguration
Exception Message:
GeometryConfiguration not found in process history.Searched for process HLT and label XMLIdealGeometryESSource@.
----- End Fatal Exception -------------------------------------------------
15-Apr-2020 04:03:37 CEST  Closed file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_10_6_0_patch2/RelValSingleMuPt10/GEN-SIM/106X_upgrade2023_realistic_v3_2023D41noPU-v1/10000/7377ED92-245C-CC4D-9F05-25ABB5522A08.root

=============================================

MessageLogger Summary

 type     category        sev    module        subroutine        count    total
 ---- -------------------- -- ---------------- ----------------  -----    -----

```

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/cc8_amd64_gcc8/CMSSW_11_1_X_2020-04-14-2300/unitTestLogs/SLHCUpgradeSimulations/Geometry#/

Likely, this error is related to #29151
cc: @tschuh

**Comment by silviodonato** [comment]
assign upgrade

**Comment by silviodonato** [comment]
cc: @emiglior

**Comment by silviodonato** [comment]
I attached the diff between `CMSSW_11_1_X_2020-04-14-1100` and `CMSSW_11_1_X_2020-04-14-2300` of the `edmConfigDump` of the configuration used by the unit test `SLHCUpgradeSimulations/Geometry/test/phase2_digi_reco_pixelntuple_cfg.py` 

[diff.txt](https://github.com/cms-sw/cmssw/files/4479660/diff.txt)

After #29151, we added `TrackerDTCProducer` in `L1TrackTrigger_step` with the following configuration
```
+process.TrackerDTCProducer = cms.EDProducer("trackerDTC::Producer",
+    ParamsConverter = cms.PSet(
+        WidthQoverPt = cms.int32(9),
+        WidthRowLUT = cms.int32(4)
+    ),
+    ParamsED = cms.PSet(
+        CheckHistory = cms.bool(True),
+        DataFormat = cms.string('Hybrid'),
+        InputTagCablingMap = cms.ESInputTag("GlobalTag",""),
+        InputTagGeometryConfiguration = cms.ESInputTag("XMLIdealGeometryESSource",""),
+        InputTagMagneticField = cms.ESInputTag("VolumeBasedMagneticFieldESProducer",""),
+        InputTagTTStubAlgorithm = cms.ESInputTag("TTStubAlgorithm_official_Phase2TrackerDigi_",""),
+        InputTagTTStubDetSetVec = cms.InputTag("TTStubsFromPhase2TrackerDigis","StubAccepted"),
+        InputTagTrackerGeometry = cms.ESInputTag("trackerGeometry",""),
+        InputTagTrackerTopology = cms.ESInputTag("trackerTopology",""),
+        OffsetDetIdDSV = cms.int32(1),
+        OffsetDetIdTP = cms.int32(-1),
+        OffsetLayerDisks = cms.int32(10),
+        OffsetLayerId = cms.int32(1),
+        ProcessName = cms.string('HLT'),
+        ProductBranch = cms.string('StubAccepted'),
+        ProductLabel = cms.string('XMLIdealGeometryESSource'),
+        SupportedTrackerXMLFile = cms.string('tracker.xml'),
+        SupportedTrackerXMLPSet = cms.string('geomXMLFiles'),
+        SupportedTrackerXMLPath = cms.string('Geometry/TrackerCommonData/data/PhaseII/'),
+        SupportedTrackerXMLVersions = cms.vstring(
+            'TiltedTracker613', 
+            'TiltedTracker613_MB_2019_04'
+        )
+    ),
+    ParamsFW = cms.PSet(
+        BField = cms.double(3.81120228767),
+        BFieldError = cms.double(1e-06),
+        InnerRadius = cms.double(21.8),
+        MaxPitch = cms.double(0.01),
+        OuterRadius = cms.double(112.7),
+        SpeedOfLight = cms.double(2.99792458)
+    ),
+    ParamsFormat = cms.PSet(
+        ChosenRofPhi = cms.double(55.0),
+        Disk2SRsSet = cms.VPSet(
+            cms.PSet(
+                Disk2SRs = cms.vdouble(
+                    66.7345, 71.7345, 77.5056, 82.5056, 84.8444, 
+                    89.8444, 95.7515, 100.7515, 102.475, 107.475
+                )
+            ), 
+            cms.PSet(
+                Disk2SRs = cms.vdouble(
+                    66.7345, 71.7345, 77.5056, 82.5056, 84.8444, 
+                    89.8444, 95.7515, 100.7515, 102.475, 107.475
+                )
+            ), 
+            cms.PSet(
+                Disk2SRs = cms.vdouble(
+                    65.1317, 70.1317, 75.63, 80.63, 83.9293, 
+                    88.9293, 94.6316, 99.6316, 102.475, 107.475
+                )
+            ), 
+            cms.PSet(
+                Disk2SRs = cms.vdouble(
+                    65.1317, 70.1317, 75.63, 80.63, 83.9293, 
+                    88.9293, 94.6316, 99.6316, 102.475, 107.475
+                )
+            ), 
+            cms.PSet(
+                Disk2SRs = cms.vdouble(
+                    65.1317, 70.1317, 75.63, 80.63, 83.9293, 
+                    88.9293, 94.6316, 99.6316, 102.475, 107.475
+                )
+            )
+        ),
+        DiskZs = cms.vdouble(131.18, 155.0, 185.34, 221.619, 265.0),
+        LayerRs = cms.vdouble(
+            24.8656, 37.1678, 52.27, 68.7, 86.0, 
+            110.8
+        ),
+        MaxEta = cms.double(2.5),
+        MinPt = cms.double(2.0),
+        NumLayers = cms.int32(4),
+        NumRingsPS = cms.vint32(11, 11, 8, 8, 8),
+        RangesAlpha = cms.vdouble(0.0, 0.0, 0.0, 1024.0),
+        RangesR = cms.vdouble(7.5, 7.5, 120.0, 0.0),
+        RangesZ = cms.vdouble(240.0, 240.0, 7.5, 7.5),
+        WidthsAlpha = cms.vint32(0, 0, 0, 4),
+        WidthsBend = cms.vint32(3, 4, 3, 4),
+        WidthsPhi = cms.vint32(14, 17, 14, 14),
+        WidthsR = cms.vint32(7, 7, 12, 7),
+        WidthsZ = cms.vint32(12, 8, 7, 7)
+    ),
+    ParamsRouter = cms.PSet(
+        EnableTruncation = cms.bool(False),
+        FreqDTC = cms.double(360.0),
+        NumFramesInfra = cms.int32(6),
+        NumRoutingBlocks = cms.int32(2),
+        SizeStack = cms.int32(64),
+        TMP_TFP = cms.int32(18)
+    ),
+    ParamsTTStubAlgo = cms.PSet(
+        BaseWindowSize = cms.double(0.5),
+        CheckHistory = cms.bool(False),
+        Label = cms.string('TTStubAlgorithm_official_Phase2TrackerDigi_'),
+        Process = cms.string('HLT')
+    ),
+    ParamsTracker = cms.PSet(
+        BaseBend = cms.double(0.25),
+        BaseCol = cms.double(1.0),
+        BaseRow = cms.double(0.5),
+        BendCut = cms.double(1.3125),
+        FreqLHC = cms.double(40.0),
+        NumDTCsPerRegion = cms.int32(24),
+        NumModulesPerDTC = cms.int32(72),
+        NumOverlappingRegions = cms.int32(2),
+        NumRegions = cms.int32(9),
+        TMP_FE = cms.int32(8),
+        WidthBend = cms.int32(6),
+        WidthCol = cms.int32(5),
+        WidthRow = cms.int32(11)
+    )
+)

```

**Comment by silviodonato** [comment]
I think the problem is due to https://github.com/cms-sw/cmssw/pull/29151/files#diff-e849ebacbb10dcdc5322db60df5c7a9aR26
Replacing the line to
`ProcessName                   = cms.string    ( "" ),                                             #`
should solve the problem

**Comment by silviodonato** [comment]
assign l1

**Comment by tschuh** [comment]
Ups, in  L1Trigger/TrackerDTC/python/Producer_Defaults_cfi.p L25 CheckHistory should be False by default. Are you able to change that or shall I make a PR?

**Comment by silviodonato** [comment]
please make a PR, thanks

**Comment by silviodonato** [comment]
fixed by  #29478

**Comment by silviodonato** [comment]
#29478 solved the problem
